src/Microsoft.Framework.PackageManager/Restore/NuGet/HttpSource.cs

```diff
@@ -129,7 +129,7 @@ internal class HttpSource
             {
                 HttpStatusCode statusCode;
 
-                using (var response = await _client.SendAsync(request))
+                using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead))
                 {
                     response.EnsureSuccessStatusCode();
```

Comment:
Yeah, I wanted to avoid deeply nested `using` statements here. With `FileOptions.DeleteOnClose` option, the temporary file will always be deleted if status code is not successful or response content is invalid